### PR TITLE
[Security] add & update doc entries on AbstractVoter implementation

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes (https://github.com/symfony/symfony/pull/11183) |
| Applies to | 2.6+ |
| Fixed tickets | - |

This PR finishes #4257.

### CODE REVIEW

File: cookbook/security/voters.rst [hunk 1]
```diff
@@ -92,6 +92,13 @@ the security layer. This can be done easily through the service container.
     methods in your implementation of the ``vote()`` method and return ``ACCESS_ABSTAIN``
     if your voter does not support the class or attribute.
 
+
```
Comment (xabbuh) [inline cookbook/security/voters.rst]: please remove one of the blank lines

File: cookbook/security/voters.rst [hunk 2]
```diff
@@ -92,6 +92,13 @@ the security layer. This can be done easily through the service container.
...
+.. tip::
+
+    An
+    :class:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\AbstractVoter`
+    is provided to cover the common use cases when implementing security voters.
```
Comment (xabbuh) [inline cookbook/security/voters.rst]: What about linking "AbstractVoter" to the cookbook?

File: cookbook/security/voters_data_permission.rst [hunk 3]
```diff
@@ -61,90 +69,74 @@ edit a particular object. Here's an example implementation:
...
     }
 
 That's it! The voter is done. The next step is to inject the voter into
 the security layer.
 
+To recap, here's what's expected from the three abstract methods:
+
+:method:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\AbstractVoter::getSupportedClasses`
+    It tells Symfony that your voter should be called whenever an object of one
+    of the given classes is passed to ``isGranted()``  For example, if you return
```
Comment (xabbuh) [inline cookbook/security/voters_data_permission.rst]: missing period after `isGranted()`

File: cookbook/security/voters_data_permission.rst [hunk 4]
```diff
@@ -61,90 +69,74 @@ edit a particular object. Here's an example implementation:
...
+    ``array('AppBundle\Model\Product')``, Symfony will call your voter when a
+    ``Product`` object is passed to ``isGranted()``.
+
+:method:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\AbstractVoter::getSupportedAttributes`
+    It tells Symfony that your voter should be called whenever one of these
+    strings is passes as the first argument to ``isGranted()``. For example, if
```
Comment (xabbuh) [inline cookbook/security/voters_data_permission.rst]: [...] is passed as [...]

File: cookbook/security/voters.rst (hunk 2 above)
Comment (javiereguiluz) [inline cookbook/security/voters.rst]: I'm afraid that I don't understand this suggestion. Which page should I link to?

Comment (xabbuh) [inline cookbook/security/voters.rst]: I thought about linking to `/cookbook/security/voters_data_permission.rst`. What do you think about that?

Comment (javiereguiluz) [inline cookbook/security/voters.rst]: OK. I understand it now. I've just added the link. Thanks.

**Comment by xabbuh** [comment]
:+1: reads very well, Javier

**Comment by weaverryan** [comment]
Thank you VERY much for finishing this off Javier! I'm thrilled to have this.

I made some tweaks at sha: e46f02e2fe8c0bec539f3736d7ee5ef371c2102c - the only material one being a change to switch-case - I found this more readable than the if statement - especially because I typically separate out my different attribute cases very clearly.

Cheers!

**Pull request closed without merging**